replications/internal/queue_management.go

```diff
@@ -126,3 +132,106 @@ func (qm *durableQueueManager) UpdateMaxQueueSize(replicationID platform.ID, max
 
 	return nil
 }
+
+// StartReplicationQueues updates the durableQueueManager.replicationQueues map, fully removing any partially deleted
+// queues (present on disk, but not tracked in sqlite), opening all current queues, and logging info for each.
+func (qm *durableQueueManager) StartReplicationQueues(trackedReplications *influxdb.Replications) error {
+	errOccurred := false
+
+	for _, r := range trackedReplications.Replications {
+		// Re-initialize a queue struct for each replication stream from sqlite
+		queue, err := durablequeue.NewQueue(
+			filepath.Join(qm.queuePath, r.ID.String()),
+			r.MaxQueueSizeBytes,
+			durablequeue.DefaultSegmentSize,
+			&durablequeue.SharedCount{},
+			durablequeue.MaxWritesPending,
+			func(bytes []byte) error {
+				return nil
+			},
+		)
+
+		if err != nil {
+			qm.logger.Error("failed to initialize replication stream durable queue", zap.Error(err))
+			errOccurred = true
+		}
+
+		// Open and map the queue struct to its replication ID
+		if err := queue.Open(); err != nil {
+			qm.logger.Error("failed to open replication stream durable queue", zap.Error(err), zap.String("id", r.ID.String()))
+			errOccurred = true
+		} else {
+			qm.replicationQueues[r.ID] = queue
+			qm.logger.Info("Opened replication stream", zap.String("id", r.ID.String()), zap.String("path", queue.Dir()))
+		}
+	}
+
+	if errOccurred {
+		return fmt.Errorf("startup tasks for replications durable queue management failed, see server logs for details")
+	}
+
+	// Get contents of replicationq directory
+	entries, err := os.ReadDir(qm.queuePath)
+	if err != nil {
+		return err
+	}
+
+	for _, entry := range entries {
+		// Skip over non-relevant entries (must be a dir named with a replication ID)
+		if !entry.IsDir() {
+			continue
+		}
+
+		id, err := platform.IDFromString(entry.Name())
+		if err != nil {
+			continue
+		}
+
+		// Partial delete found, needs to be fully removed
+		if qm.replicationQueues[*id] == nil {
+			// Initialize queue struct for existing queue on disk
+			newQueue, err := durablequeue.NewQueue(
```

Comment:
Do we need to create it in order to remove it? Could we `os.RemoveAll` the directory instead?